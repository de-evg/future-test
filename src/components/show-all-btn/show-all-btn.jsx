import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { fetchFullUsers } from "../../store/api-action";
import {ActionCreator} from "../../store/action";

const ShowAllBtn = ({ loadFullData, setIsLoadingStatus }) => {
  const handleClick = useCallback(() => {
    setIsLoadingStatus(true);
    loadFullData();
  }, [loadFullData, setIsLoadingStatus]);

  return (
    <button className="main-menu__button button" onClick={handleClick}>
      Показать все
    </button>
  );
};

ShowAllBtn.propTypes = {
  loadFullData: PropTypes.func.isRequired,
  setIsLoadingStatus: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  loadFullData() {
    dispatch(fetchFullUsers());
  },
  setIsLoadingStatus(status) {
    dispatch(ActionCreator.updateLoadingStatus(status))
  }
});

export default connect(null, mapDispatchToProps)(ShowAllBtn);
