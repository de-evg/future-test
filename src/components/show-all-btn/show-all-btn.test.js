import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { ShowAllBtn } from "./show-all-btn";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

it(`Reder ShowAllBtn correctly`, () => {
  const wrapper = shallow(
    <ShowAllBtn loadFullData={() => {}} setIsLoadingStatus={() => {}} />
  );
  const tree = shallowToJson(wrapper);
  expect(tree).toMatchSnapshot();
});
