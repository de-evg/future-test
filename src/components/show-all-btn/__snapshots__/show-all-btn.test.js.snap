// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reder ShowAllBtn correctly 1`] = `
<button
  className="main-menu__button button"
  onClick={[Function]}
>
  Показать все
</button>
`;
